libass/ass.c

```diff
@@ -301,25 +317,28 @@ static long long string2timecode(ASS_Library *library, char *p)
 	} else if (ass_strcasecmp(tname, #name) == 0) { \
 		target->name = lookup_style(track, token);
 
+// skip spaces in str beforehand, or trim leading spaces afterwards
+static inline void advance_token_pos(char **const str,
+                                     char **const start, char **const end)
+{
+    *start = *str;
+    *end   = *start;
+    while(**end != '\0' && **end != ',') ++*end;
+    *str = *end + (**end == ',');
+    rskip_spaces(end, *start);
+}
+
 static char *next_token(char **str)
 {
-    char *p = *str;
+    char *p;
     char *start;
-    skip_spaces(&p);
-    if (*p == '\0') {
-        *str = p;
+    skip_spaces(str);
+    if (**str == '\0') {
         return 0;
     }
-    start = p;                  // start of the token
-    for (; (*p != '\0') && (*p != ','); ++p) {
-    }
-    if (*p == '\0') {
-        *str = p;               // eos found, str will point to '\0' at exit
-    } else {
-        *p = '\0';
-        *str = p + 1;           // ',' found, str will point to the next char (beginning of the next token)
-    }
-    rskip_spaces(&p, start);    // end of current token: the first space character, or '\0'
+
+    advance_token_pos(str, &start, &p);
```

Comment:
With this change `next_token` no longer replaces **all** `','` with `'\0'`, only if they terminate a token.
As far as I can tell, this was superfluous and removing it should not have any negative impact.